docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -6,6 +6,191 @@ title: Data Grid - Group & Pivot
+
+### Grouping columns
+
+#### Single grouping column
+
+By default, the grid will create only one grouping column even if you have several grouping criteria.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingSingleGroupingCol.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Multiple grouping column
+
+To have a grouping column for each grouping criteria, set the `rowGroupingColumnMode` prop to `multiple`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingMultipleGroupingCol.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Custom grouping column
+
+Use the `groupingColDef` prop to customize the rendering of the grouping column. You can override any property of the `GridColDef` interface except the `field`, the `type` and the properties related to the edition.
+
+If you want to apply your overrides to every grouping column, use the object format of `groupingColDef`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingCustomGroupingColDefObject.js", "bg": "inline", "defaultCodeOpen": false}}
+
+If you want to only override properties of certain grouping columns or to apply different overrides based on the current grouping criteria, use the callback format of `groupingColDef`.
+It will be called for each grouping column with the fields of the columns used to build it.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Show values for the leaves
+
+By default, the leaves nodes don't render anything for their grouping cell.
+
+If you want to display some value, you can provide a `leafField` property to the `groupingColDef`.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingLeafWithValue.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Hide the descendant count
+
+You can use the `hideDescendantCount` property of the `groupingColDef` to hide the amount of descendant of a grouping row.
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingHideDescendantCount.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Complex grouping value
+
+In most scenarios, when you need to handle complex values, you provide a `valueGetter` property to your column definition.
+But sometimes, you need to keep the object format for the `renderCell` property and thus can not convert it to a serializable value in `valueGetter`.
+
+You can then provide a `keyGetter` property in your column definition to convert this object into a serializable value.
+
+```ts
+const columns: GridColumns = [
+  {
+    field: 'composer',
+    keyGetter: (params) => params.value.name,
+  },
+  // ...
+];
+```
+
+{{"demo": "pages/components/data-grid/group-pivot/RowGroupingKeyGetter.js", "bg": "inline", "defaultCodeOpen": false}}
+
+If your column also have a `valueGetter` property, the value passed to the `keyGetter` property will be the one returned by `valueGetter`.
```

Comment:
*as a note: if your column also have a `valueGetter` property, its value is used as the parameter for the `keyGetter` method.